### PR TITLE
get aws information from link local address

### PR DESCRIPTION
**Pull request by ramaraochavali**
AWS provides locality information on link local IP that can be accessed locally on each node. https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instancedata-data-retrieval.html instead of using metadata service.

This is simpler to use and does not depend on restricted network access.

This PR refactor AWS platform code to use that approach.

Fixes https://github.com/istio/istio/issues/37832

I have tested this in one of our clusters accessing those urls and here is the output

```
bash-4.2$ curl http://169.254.169.254/latest/meta-data/instance-id
i-0a26ab14a
bash-4.2$ curl http://169.254.169.254/latest/meta-data/placement/region
us-west-2
bash-4.2$ curl http://169.254.169.254/latest/meta-data/placement/availability-zone
us-west-2b
```

- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [X] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

### CODE REVIEW

File: pkg/bootstrap/platform/aws.go [hunk 1]
```diff
@@ -15,83 +15,66 @@
 package platform
 
 import (
-	"errors"
-	"io/fs"
-	"os"
-	"strings"
-
-	"github.com/aws/aws-sdk-go/aws"
```
Comment (LithiumSR) [inline pkg/bootstrap/platform/aws.go:23]: Can the dependency on github.com/aws/aws-sdk-go be removed from go.mod now that is no longer used in aws.go and aws_test.go?

Comment (ramaraochavali) [reply pkg/bootstrap/platform/aws.go:23]: Yes. It is removed now.

File: pkg/bootstrap/platform/aws.go [hunk 2]
```diff
@@ -15,83 +15,66 @@
...
 )
 
 const (
 	AWSRegion           = "aws_region"
 	AWSAvailabilityZone = "aws_availability_zone"
 	AWSInstanceID       = "aws_instance_id"
-	AWSAccountID        = "aws_account_id"
 )
 
+var awsMetadataURL = "http://169.254.169.254/latest/meta-data"
```
[the remark this answers is not in the document]
Comment (ramaraochavali) [reply pkg/bootstrap/platform/aws.go:32]: Good point. I am making change to consider ipv6 also

[the remark this answers is not in the document]
Comment (costinm) [reply pkg/bootstrap/platform/aws.go:32]: Could you make this an env variable, for testing ? GCP std library uses GCE_METADATA_HOST. 

I would check aws library if they define a similar setting.

**Review by costinm — APPROVED** [review]
Great change. 

We should probably have a common library with GCP - same MDS IP but there are differences in the URL format. How about Azure ?

**Comment by LithiumSR** [comment]
Thanks a lot @ramaraochavali for the fix!
Would be possible to backport this commit and the IPv6 support follow-up (#37874) to the 1.13 branch as that is where the #37832 issue started showing up?

**Comment by ramaraochavali** [comment]
@LithiumSR Backport to release 1.3 https://github.com/istio/istio/pull/37886

**Pull request merged**